### PR TITLE
New interface to implement DROOLS-995. Exposes rule properties for ru…

### PR DESCRIPTION
**Pull request by DuncanDoyle**
…ntime rule inspection.

Initial idea which implements DROOLS-995. Exposes most rule properties for runtime rule inspection.

Note that I couldn't expose the getTimer() method via this interface, as it returns a Timer, which is a Drools internal class.

### CODE REVIEW

**Comment by mrietveld** [comment]
@DuncanDoyle Test? If only to show use and enforce backwards compatibility? Thanks.

**Comment by DuncanDoyle** [comment]
@mrietveld: I looked at that (before I created the PR), and the kie-internal project doesn't really contain tests (other than the KieMetaTest). This PR is actually part of another on: https://github.com/droolsjbpm/drools/pull/579

That PR changes the interface of RuleImpl from Rule to InternalRule (the interface added in this PR). Other than that I didn't change much, I'm just exposing the already existing methods via this interface so we can do more runtime rule inspection, and not having to depend on the Impl class (and I added two small methods to expose the Salience of a rule without the need of the Salience class in kie-internal).

I've tested the other PR using the standard Maven build (mvn clean install), and the new intermediate interface seems to work fine (well, no test-failures AFAICT).

So, long story short, I'm not entirely sure how to write a test for the interface in this PR. Any suggestions?

**Comment by mrietveld** [comment]
Add or extend a test in drools-core (as part of droolsjbpm/drools#579) that uses the 2 new methods (getSalienceValue, isSalienceDynamic). 

Would that be possible?

**Comment by DuncanDoyle** [comment]
OK, will do that. Will add a comment here when I've added those tests.

**Comment by DuncanDoyle** [comment]
Unit-tests for 'getSalienceValue' and 'isSalienceDynamic' added to this PR: https://github.com/droolsjbpm/drools/pull/579

**Comment by mariofusco** [comment]
Merged by https://github.com/droolsjbpm/droolsjbpm-knowledge/commit/044c3a0b8a0f7c93100339f142397988c16d7dde
